tests/integration/suite/healthz/placement.go

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthz
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	procplace "github.com/dapr/dapr/tests/integration/framework/process/placement"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(placement))
+}
+
+// placement tests that Dapr responds to healthz requests.
+type placement struct {
+	proc *procplace.Placement
+}
+
+func (d *placement) Setup(t *testing.T) []framework.Option {
+	d.proc = procplace.New(t)
+	return []framework.Option{
+		framework.WithProcesses(d.proc),
+	}
+}
+
+func (d *placement) Run(t *testing.T, ctx context.Context) {
+	assert.Eventuallyf(t, func() bool {
+		conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", d.proc.HealthzPort))
+		if err != nil {
+			return false
+		}
+		require.NoError(t, conn.Close())
+		return true
+	}, time.Second*5, 100*time.Millisecond, "healthz port %d not ready", d.proc.HealthzPort)
+
+	reqURL := fmt.Sprintf("http://127.0.0.1:%d/healthz", d.proc.HealthzPort)
+
+	assert.Eventually(t, func() bool {
+		req, err := http.NewRequestWithContext(ctx, http.MethodGet, reqURL, nil)
```

Comment:
Context with a timeout please, since using the default client

Reply:
The context is scoped to the test case which has a timeout. It don't see the need to wrap it again with another timeout.

Reply:
I responded to Shubham here too: https://github.com/dapr/dapr/pull/6524#discussion_r1232501908

The timeout is the timeout for the entire test. If a request hanged for any reason, it would take up the entire timeout for the test, and it would never retry in the `Eventually` block.

Reply:
Is this not the right behavior? I want the test to fail if the connection doesn't work. I'm retrying here only to allow placement to return with a successful healthy=false response, and then eventually a healthy=true response.

Reply:
Ok then if this the behavior you want, sounds good :)